ddtrace/contrib/kafka/patch.py

```diff
@@ -123,6 +131,10 @@ def traced_poll(func, instance, args, kwargs):
         span.set_tag_str(kafkax.RECEIVED_MESSAGE, str(message is not None))
         span.set_tag_str(kafkax.GROUP_ID, instance._group_id)
         if message is not None:
+            headers = {header[0]: header[1] for header in (message.headers() or [])}
+            if config._data_streams_enabled:
+                ctx = pin.tracer.data_streams_processor.decode_pathway(headers.get(PROPAGATION_KEY, None))
+                ctx.set_checkpoint(["direction:in", "group:"+instance._group_id, "topic:"+message.topic(), "type:kafka"])
```

Comment:
It seems like we're not propagating the context anywhere. This will still work, but we not gonna be able to get true end-to-end latency. Should we consider having some sort of  context tracking within `pin.tracer.data_streams_processor`? Maybe using a `ContextVar`?

Reply:
Added context propagation as discussed.